_docs_integrate/error-codes.md

```diff
@@ -24,76 +24,167 @@ required_by:
+| {% include anchor a="error.runtime.attributes.isNotPeerSharedAttribute" %}                                          | The given Attribute is not a [peer shared Attribute]({% link _docs_integrate/data-model-overview.md %}#localattribute).                                                                                                                                                                                                  |
+| {% include anchor a="error.runtime.attributes.isNotRepositoryAttribute" %}                                          | The given Attribute is not a [RepositoryAttribute]({% link _docs_integrate/data-model-overview.md %}#localattribute).                                                                                                                                                                                                    |
+| {% include anchor a="error.runtime.attributes.isNotThirdPartyOwnedRelationshipAttribute" %}                         | The given Attribute is not a third party owned [RelationshipAttribute]({% link _docs_integrate/data-model-overview.md %}#localattribute).                                                                                                                                                                                |
+| {% include anchor a="error.runtime.attributes.noPreviousVersionOfRepositoryAttributeHasBeenSharedWithPeerBefore" %} | No previous version of the given [RepositoryAttribute]({% link _docs_integrate/data-model-overview.md %}#localattribute) has been shared with this peer before. If you wish to execute an initial sharing of this Attribute, [use 'ShareRepositoryAttribute']({% link _docs_integrate/share-attributes-with-peer.md %}). |
+| {% include anchor a="error.runtime.attributes.repositoryAttributeHasAlreadyBeenSharedWithPeer" %}                   | The given [RepositoryAttribute]({% link _docs_integrate/data-model-overview.md %}#localattribute) has already been shared with this peer.                                                                                                                                                                                |
+| {% include anchor a="error.runtime.cacheEmpty" %}                                                                   | The `cache` of the requested entity is empty.                                                                                                                                                                                                                                                                            |
```

Comment:
We use the cache only internally, it can be written normally

Reply:
It can be done later, just put the comment as a reminder

Reply:
No I will change all this things tomorrow morning and then we can merge this.

Reply:
Please remove this error completely. The integrator will never see it.

Reply:
this should not be an error code in the runtime at all...